digitalforms-api/src/main/java/ca/bc/gov/open/digitalformsapi/viirp/controller/DfPayloadsApiDelegateImpl.java

```diff
@@ -118,13 +112,27 @@ public ResponseEntity<PostDFPayloadServiceResponse> dfpayloadsNoticeNoCorrelatio
 
 		logger.info("Heard a call to the endpoint 'dfpayloadsNoticeNoCorrelationIdDelete' with noticeNo " + noticeNo);
 		
-		// TODO - need to pass the params to the ORDS call to delete notice no data. 
-		// TODO - need to set response type (or exception) based on ORDS response. 
-
-		PostDFPayloadServiceResponse resp = new PostDFPayloadServiceResponse();
-		resp.setStatusMessage("success");
-
-		return new ResponseEntity<>(resp, HttpStatus.OK);
-
+		ca.bc.gov.open.pssg.rsbc.digitalforms.ordsclient.api.model.PostDFPayloadServiceResponse responseFromOrds;
+		try {
+			// ORDS call to delete DF Payload based on notice no
+			responseFromOrds = dfPayloadService.deleteDFPayload(noticeNo, correlationId);
+		} catch (ApiException e) {
+			String msg = "ERROR deleting DF Payload from ORDS with noticeNo: " + noticeNo + ". Message: " + e.getMessage() + " ORDS Response Status Code: " + e.getCode();
+			e.printStackTrace();
+			logger.error(msg, e);
+			throw new DigitalFormsException(e.getMessage(), e);
+		}
+		
```

Comment:
Hi Burak. The following should probably change. I'm hoping Kyle will be sending us back proper HTTP status codes to indicate success or failure from ORDS in which case we would catch an APIException. This Exception type is thrown in the case of 40x or 50x status codes. The extra benefit is that we don't have to sniff through the response looking for a message that may or may not exist. See the bit of coding I recently implemented for the DfPayloadsApiDelegateImpl.java. Thanks